File: GrowthWare_VB/GrowthWareFrameworkModel/Profiles/MMessageProfile.vb
Imports GrowthWare.Framework.ModelObjects.Base.Interfaces
Imports GrowthWare.Framework.ModelObjects.Base
Imports System.Reflection

Namespace ModelObjects
    <Serializable(), CLSCompliant(True)> _
    Public Class MMessageProfile
		Inherits MFormatter
        Implements IProfile, IMessageProfile

#Region "Member Properties"
		Private m_SE_SEQ_ID As Integer = 1
		Private m_Description As String = String.Empty
		Private m_ErrorCode As Integer = -1
		Private m_Title As String = String.Empty
		Private m_FormatAsHTML As Boolean = False
#End Region

#Region "Protected Methods"
		Private Overloads Sub Init(ByVal dr As DataRow)
			On Error Resume Next
			MyBase.Init(dr)
			m_ID = CInt(dr("MESSAGE_SEQ_ID"))
			m_SE_SEQ_ID = CInt(dr("SE_SEQ_ID"))
			m_Title = CStr(dr("TITLE")).Trim
			m_Description = CStr(dr("DESCRIPTION")).Trim
			m_FormatAsHTML = CBool(dr("FORMAT_AS_HTML"))
			m_Body = CStr(dr("BODY"))
		End Sub
#End Region

#Region "Public Methods"
		''' <summary>
		''' Will return a message profile with the default vaules
		''' </summary>
		''' <remarks></remarks>
		Public Sub New()

		End Sub

		Public Sub New(ByVal dr As DataRow)
			Init(dr)
		End Sub

		Public Overridable Sub FormatBody() Implements IMessageProfile.FormatBody
			Dim myPropertyInfo As PropertyInfo() = Me.GetType.GetProperties()
			Dim myPropertyItem As PropertyInfo
			For Each myPropertyItem In myPropertyInfo
				Dim pValue As Object = myPropertyItem.GetValue(Me, Nothing)
				m_Body = m_Body.Replace("<" & myPropertyItem.Name & ">", pValue.ToString)
			Next
		End Sub
#End Region

#Region "Public Properties"
		Public Property SE_SEQ_ID() As Integer
			Get
				Return m_SE_SEQ_ID
			End Get
			Set(ByVal value As Integer)
				m_SE_SEQ_ID = value
			End Set
		End Property

		Public Property Title() As String Implements IMessageProfile.Title
			Get
				Return m_Title
			End Get
			Set(ByVal value As String)
				m_Title = value.Trim
			End Set
		End Property

		Public Property Description() As String
			Get
				Return m_Description
			End Get
			Set(ByVal value As String)
				m_Description = value.Trim
			End Set
		End Property

		Public Property FormatAsHTML() As Boolean Implements IMessageProfile.FormatAsHTML
			Get
				Return m_FormatAsHTML
			End Get
			Set(ByVal value As Boolean)
				m_FormatAsHTML = value
			End Set
		End Property

		Public Property BODY() As String Implements IMessageProfile.Body
			Get
				Return m_Body
			End Get
			Set(ByVal value As String)
				m_Body = value
			End Set
		End Property
#End Region
    End Class
End Namespace

File: GrowthWare_CS/GrowthWareFrameworkModel/Profiles/MMessageProfile.cs
using System;
using System.Data;
using System.Reflection;
using GrowthWare.Framework.Interfaces;
using GrowthWare.Framework.ModelObjects.Base;
using GrowthWare.Framework.ModelObjects.Base.Interfaces;

namespace GrowthWare.Framework.ModelObjects
{
	[Serializable(), CLSCompliant(true)]
	public class MMessageProfile : MFormatter
	{
#region "Member Properties"
		private int m_SE_SEQ_ID = 1;
		private string m_Description = string.Empty;
		private string m_Title = string.Empty;
		private bool m_FormatAsHTML = false;
#endregion

#region "Protected Methods"
		private new void init(DataRow Datarow)
		{
			base.init(Datarow);
			base.setInt(ref m_ID, ref Datarow, "MESSAGE_SEQ_ID");
			base.setInt(ref m_SE_SEQ_ID, ref Datarow, "SE_SEQ_ID");
			base.setString(ref m_Title, ref Datarow, "TITLE");
			base.setString(ref m_Description, ref Datarow, "DESCRIPTION");
			base.setBool(ref m_FormatAsHTML, ref Datarow, "FORMAT_AS_HTML");
		}
#endregion

#region "Public Methods"
		/// <summary>
		/// Will return a message profile with the default vaules
		/// </summary>
		/// <remarks></remarks>

		public MMessageProfile()
		{
		}

		public MMessageProfile(DataRow dr)
		{
			init(dr);
		}

		public virtual void FormatBody()
		{
			PropertyInfo[] myPropertyInfo = this.GetType().GetProperties();
			PropertyInfo myPropertyItem = null;
			foreach (PropertyInfo myPropertyItem_loopVariable in myPropertyInfo) {
				myPropertyItem = myPropertyItem_loopVariable;
				object pValue = myPropertyItem.GetValue(this, null);
				base.m_Body = base.m_Body.Replace("<" + myPropertyItem.Name + ">", pValue.ToString());
			}
		}
#endregion

#region "Public Properties"
		public int SE_SEQ_ID {
			get { return m_SE_SEQ_ID; }
			set { m_SE_SEQ_ID = value; }
		}

		public string Title {
			get { return m_Title; }
			set { m_Title = value.Trim(); }
		}

		public string Description {
			get { return m_Description; }
			set { m_Description = value.Trim(); }
		}

		public bool FormatAsHTML {
			get { return m_FormatAsHTML; }
			set { m_FormatAsHTML = value; }
		}

		public string Body {
			get { return m_Body; }
			set { m_Body = value.Trim(); }
		}

#endregion
	}
}
